source/fundamentals/serialization/poco.txt

```diff
@@ -261,6 +261,167 @@ The following code sample maps the ``Identifier`` property to the
    The ``_id`` field mapping logic described in this section only applies to the
    root document and does not apply to nested documents.
 
+Specify an ID Generator
+``````````````````````
+
+Every document in a MongoDB collection must have a unique ID. When you serialize an object
+to a collection, if the ``Id`` property contains the default
+value for its data type (usually ``null``), the {+driver-short+} doesn't serialize the
+default value. Instead, the driver generates a unique ID value for the property.
+
+When you apply the ``[BsonId]`` attribute to a property, you can pass the ``IdGenerator``
+argument to specify the type of ID value the driver generates for the corresponding
+MongoDB field. The following table describes the ID generators available in the
+{+driver-short+}:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 75
+
+   * - Id Field Data Type
+     - How to Use
+
+   * - ``Guid``
+     - To use the COMB algorithm to generate a unique ``Guid`` value, apply the
+       ``[BsonId(IdGenerator = typeof(CombGuidGenerator))]`` attribute to the ``Id``
+       property, as shown in the following example:
+
+       .. code-block:: csharp
+          :copyable: true
+
+          public class House
+          {
+             [BsonId(IdGenerator = typeof(CombGuidGenerator))]
+             public Guid Id { get; set; }
+          }
+
+       To generate a unique ``Guid`` value without the COMB algorithm, don't apply
+       the preceding attribute to the ``Id`` property. The driver automatically uses
+       the ``GuidGenerator`` type to generate a unique value for the ``Id`` field.
+
+       .. code-block:: csharp
+          :copyable: true
+
+          public class House
+          {
+             public Guid Id { get; set; }
+          } 
+
+   * - ``ObjectId``
+     - The driver automatically uses the ``ObjectIdGenerator`` type for ``Id`` fields with
+       the ``ObjectId`` data type, such as the one in the following code example:
+
+       .. code-block:: csharp
+          :copyable: true
+
+          public class House
+          {
+             public ObjectId Id { get; set; }
+          } 
+
+   * - ``string``
+     - To serialize the ``Id`` property as a ``string``, apply the
```

Comment:
Applying the `[BsonId]` attribute is NOT what causes this property to be serialized as a string.

All strings are serialized as strings UNLESS the `[BsonElement]` attribute specifies a different representation.

The ONLY thing this `[BsonId]` attribute is specifying is which id generator to use (beside identifying this as the ID field of course).

Reply:
After looking at the code for the Generator classes, I see that I've been conflating the ID generation and serialization steps.